### ISSUE TITLE
Reimplement PlayerVoiceIn and PlayerVoiceOut or similar

### ISSUE THREAD

**Issue by SleepyMode** [Facepunch/sbox-issues#1050]
**What can't you do?**
Modify binary voice data of players speaking to implement, for example, radio or cellphone effects.

**How would you like it to work?**
Hopefully, the same way that was talked about in the March 2021 blog, with functions that allow you to modify the binary data of incoming/outgoing voice data.

**What have you tried?**
Searched pretty far and couldn't find these functions or their equivalents currently.

**Additional context**
Blog post [HERE](https://sbox.facepunch.com/news/march2021#VoiceChat).
![](https://files.facepunch.com/layla/1b2911b1/devenv_M388DwkVGH2.png)

**Comment by Jake-NSW** [comment]
Would love to see this

**Comment by garrynewman** [comment]
I'm not sure if editing bytes is the way to go about adding audio effects to playback of a sound though?

**Comment by Jake-NSW** [comment]
Not really that, I was thinking more control over voice chat. Like exposing Voice.Enabled so I could be like hey I got this radio in my hand and I want to talk into it by holding Mouse1 and I want everyone on my team to hear it, but also play it in local space so people near me can hear it to. 

Also being able to hear voice in a 3D space using audio streaming using the voice data would be cool as too,

**Comment by garrynewman** [comment]
Yeah these are different issues, but I think VR generally has open mic so we'd need these options eventually

**Comment by Alf21** [comment]
Seems to be pretty useful. Currently tried to implement a proximity chat / 3D chat, but as we don't have access to modify the voicedata, we can just decide whether a certain player is able to hear another player. Currently, there is no way to implement a specified volume drop based on various factors such as distance (if I don't missed anything)
